enhancements/microshift/audit-log-configuration-options.md

```diff
@@ -0,0 +1,249 @@
+- Exposing the apiserver's audit-log-path, which allows users to set a custom log location, would hinder sos report gathering. Instead, users should replace `/var/log/kube-apiserver` with a symlink the desired path.  Sos will follow the symlink and collect logs by default.
+
+### Drawbacks
+
+- N/A
+
+## Open Questions
+
+- Custom log paths: Can sos collect reports from dynamic paths? If not, can we symlink the log directory to as a means of faking a static path while writing logs to a custom location?  If so, can sos follow these symlinks?
+
+    - SOS can traverse symlinks to collect log files from a target directory.  The link itself and the target path are included in the tarball, meaning support can easily locate audit logs via the default location (`/var/log/kube-apiserver`).
+
+- Should Microshift support a `minFreeStorage` field that would be used to determine whether an acceptable amount of space exists at the log file path? This is not a value recognized by the apiserver, but it could be used by Microshift to check if the system is in an acceptable state to boot into.  If the free space is lower than the value, MicroShift would not start and log a fatal error to document the reason. If unset, MicroShift should deploy regardless of the available storage.
+
+    - Providing additional fields and internal handling logic is outside the scope of this EP. If this is a useful feature, it should be documented in a bespoke EP.
+
+## Test Plan
+
+* Unit tests to validate new config APIs
+* Integration tests to verify configured audit logging policies are applied properly
+
+## Graduation Criteria
+
+* Upstream code, tests, docs merged
+* Downstream builds, docs updated
+* Automated CI coverage for new functionality
+* QE test plans defined
+
+### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation completed and published
+- Sufficient test coverage
+- Gather feedback from users
+- Available by default
+
+### Tech Preview -> GA
+
+- Sufficient time for feedback
+- End-to-end tests
+
+### Removing a deprecated feature
+
+- Announce deprecation and support policy of the existing feature
+- Deprecate the feature
+
+## Upgrade / Downgrade Strategy
+
+- The API must be backwards compatible for y-stream version
+
+## Operational Aspects of API Extensions
+
+- **Disk capacity exceeded**: MicroShift and the apiserver do not assess existing storage capacity for any logs, including audit logs.  As audit logs can grow quite quickly, this creates the potential for maxing out a storage device and hindering system performance.  Users must therefore consider their total storage needs, in addition to how and how often to transfer logs off-device.
+
+## Support Procedures
+
+- On startup, MicroShift should log the apiserver values and free remaining diskspace on the device containing the log files.  If the product of `maxFileSize` and `maxFiles` is greater than free storage, a warning message should be logged. 
```

Comment:
I think we might consider a watchdog thread monitoring this condition.

Reply:
ditto my comments on "Actions on Reaching Disk Capacity"

Reply:
We're giving them control over how much storage MicroShift uses for logs with this work, but users should take a holistic approach to monitoring their storage capacity and not rely on each individual component to warn them about space running out.

Reply:
+1 I'll pull mentions of logging overages.